src/component/NodeEditor/NodeEditor.tsx

```diff
@@ -611,6 +808,252 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
         this.props.onUpdateRouter(node, type, previousAction);
     }
 
+    private updateSwitchRouter(kases: CaseElementProps[]): void {
```

Comment:
moving each router's router-update method up to `NodeEditor` since, for the most part, all the info each of these methods need exists in `NodeEditor`. the only case where that isn't true is `SwitchRouter`, but that will change if we lift `SwitchRouter`'s `cases` state up to `NodeEditor`. if/when we bring in an centralized state management library, we can extract these as 'actionCreators' in Redux parlance. i'd also like to see us move away from maintaining a widget map/extracting widget state through React's `ref` API.

Reply:
This still leaves me a little sad to not have these bundled with the presentation, but I guess I get the reasoning here.

Reply:
Ya I def not ideal; I have some ideas about this